### PR TITLE
Use a prefix scan when fetching documents matching a query.

### PR DESCRIPTION
**Pull request by rsgowman**
Minor optimization (which is already present in the ts code).

http://b/36351543

### CODE REVIEW

File: Firestore/Example/Tests/Local/FSTRemoteDocumentCacheTests.m [hunk 1]
```diff
@@ -114,6 +114,7 @@ - (void)testDocumentsMatchingQuery {
   FSTDocumentDictionary *results = [self.remoteDocumentCache documentsMatchingQuery:query];
   NSArray *expected =
       @[ FSTTestDoc(@"b/1", kVersion, _kDocData, NO), FSTTestDoc(@"b/2", kVersion, _kDocData, NO) ];
+  XCTAssertEqual([results count], [expected count]);
   for (FSTDocument *doc in expected) {
     XCTAssertEqualObjects([results objectForKey:doc.key], doc);
   }
```
Comment (schmidt-sebastian) [inline Firestore/Example/Tests/Local/FSTRemoteDocumentCacheTests.m]: The TODO below is now no longer relevant, is it?

File: Firestore/Example/Tests/Local/FSTRemoteDocumentCacheTests.m [hunk 2]
```diff
@@ -114,6 +114,7 @@ - (void)testDocumentsMatchingQuery {
   FSTDocumentDictionary *results = [self.remoteDocumentCache documentsMatchingQuery:query];
   NSArray *expected =
       @[ FSTTestDoc(@"b/1", kVersion, _kDocData, NO), FSTTestDoc(@"b/2", kVersion, _kDocData, NO) ];
+  XCTAssertEqual([results count], [expected count]);
```
Comment (schmidt-sebastian) [inline Firestore/Example/Tests/Local/FSTRemoteDocumentCacheTests.m]: Can you also port the comment from JS (which is a little more descriptive than we we have here).

    // TODO(mikelehen): This just verifies that we do a prefix scan against the
    // query path. We'll need more tests once we add index support.

File: Firestore/Example/Tests/Local/FSTRemoteDocumentCacheTests.m (hunk 1 above)
Comment (rsgowman) [reply Firestore/Example/Tests/Local/FSTRemoteDocumentCacheTests.m]: Nope. Removed.

File: Firestore/Example/Tests/Local/FSTRemoteDocumentCacheTests.m (hunk 2 above)
Comment (rsgowman) [reply Firestore/Example/Tests/Local/FSTRemoteDocumentCacheTests.m]: Done.

**Pull request merged**